src/crystal/system/win32/file_descriptor.cr

```diff
@@ -6,7 +6,11 @@ module Crystal::System::FileDescriptor
   private def unbuffered_read(slice : Bytes)
     bytes_read = LibC._read(@fd, slice, slice.size)
     if bytes_read == -1
-      raise Errno.new("Error reading file")
+      if Errno.value == Errno::EBADF
```

Comment:
File descriptors can be reused - if this `if` branch can be hit, then doesn't that imply that a write may happen to the wrong file descriptor because of a race condition?

Reply:
I'm not sure how probably that reutilization.
The same check was added to `#unbuffered_write` in #6497.
Also with waj we mention that these checks might need to move to evented directly.

Reply:
> I'm not sure how probably that reutilization

Well they're sequential, so very probably.

If you `open`, `close`, `open` again, the second open will have exactly the same FD as the first.

Reply:
Yeah looks like a possible race condition, tricky one... Appears that java sets the internal "fd" to -1 before doing a close: https://github.com/openjdk-mirror/jdk7u-jdk/blob/f4d80957e89a19a29bb9f9807d2a28351ed7f7df/src/solaris/native/java/io/io_util_md.c#L93 they also synchronize the calls to close FWIW: https://github.com/openjdk-mirror/jdk7u-jdk/blob/master/src/share/classes/java/io/FileInputStream.java#L289